src/java.base/share/classes/jdk/internal/crac/mirror/Core.java

```diff
@@ -222,54 +219,30 @@ private static void checkpointRestore1(long jcmdStream) throws
             tryStartManagementAgent();
         }
 
+        assert !checkpointException.hasException() || !restoreException.hasException();
+        checkpointException.throwIfAny();
+        restoreException.throwIfAny();
+
+        final var parsedNewArguments = new ArrayList<String>();
```

Comment:
`checkpointRestore1` is already a bit long; could you extract this to another method, please?